src/theory/quantifiers/sygus/sygus_unif_strat.cpp

```diff
@@ -853,10 +885,63 @@ void SygusUnifStrategy::staticLearnRedundantOps(
         }
       }
     }
+    // do not use ITEs in occurrences of this type as a subtype of a condition
+    // enumerator type
+    if (restrictions.d_removeRetITEs)
+    {
+      const Datatype& dt =
+          static_cast<DatatypeType>(etn.toType()).getDatatype();
+      Node op = Node::fromExpr(dt[cindex].getSygusOp());
+      if (op.getKind() == kind::BUILTIN
+          && NodeManager::operatorToKind(op) == ITE)
+      {
+        for (std::pair<Node, NodeRole>& cec : etis->d_cenum)
+        {
+          // whether condition enumerator
+          if (cec.second != role_ite_condition)
+          {
+            continue;
+          }
+          // whether has etn as a subtype
+          std::vector<TypeNode> sf_types;
+          TermDbSygus* tds = d_qe->getTermDatabaseSygus();
+          tds->getSubfieldTypes(cec.first.getType(),
+                                                         sf_types);
+          if (std::find(sf_types.begin(), sf_types.end(), etn)
+              == sf_types.end())
+          {
+            continue;
+          }
+          Node fv = tds->getFreeVar(etn, 0);
```

Comment:
I think you should be iterating on the sf_types here and seeing if any of them have an ITE constructor. It seems the code here is only excluding ITE in the subterms of the type of the condition.